### ISSUE TITLE
Android Silent Push Notifications not parsing correctly

### ISSUE THREAD

**Issue by kolibrigames-ajimine-vahid** [Appboy/appboy-unity-sdk#72]
https://github.com/Appboy/appboy-unity-sdk/blob/40571203194e8fb3fa088d11246c7c5343b6543c/Assets/Plugins/Appboy/models/PushNotification.cs#L17

When we receive a silent push notification with the following:
```
{"extra":"{\"Group\":\"Test\",\"FeaturePayload\":\"{}\",\"StartImmediately\":\"true\",\"StartDate\":\"2020-09-30 00:00\",\"TargetSampleSize\":\"10000\",\"FeatureEnabled\":\"true\",\"SampleEndDate\":\"2020-09-30 00:00\",\"Type\":\"AbTest\",\"EndDate\":\"2020-10-01 00:00\",\"MainResponse\":\"LTV_day_30\",\"DisableFeatureOnEnd\":\"false\",\"SampleStartDate\":\"2020-09-29 00:00\",\"TargetResponseValue\":\"1.00\",\"SignificanceCriteria\":\"0.05\",\"UpdateNumber\":\"0\",\"Name\":\"AbTest\"}","ab_nc":"com_appboy_default_notification_channel","a":"Some Message","t":"Title","p":"0","nid":"1066395411","cid":"dGVzdF90ZXN0X2RpPWFhODU5M2UxMjkzNTc3YmQxMTJjZmViZjE3ZjcwODBi","_ab":"true","appboy_push_received_timestamp":"1601476145591"}
```

Every property in the `PushNotification` is null.

Could you look into the json deserialization? Also, if you are not doing so, it may be prudent to write unit tests for this case.

Also, it looks like `"collapse_key"` and `"notification_id"` doesn't exist in the push notification message.

----

Here is a potential solution:

I was able to solve this by using the `Newtonsoft.Json` package with the following code
```
    public PushNotification(string message) {
      if (String.IsNullOrEmpty(message)) {
        throw new ArgumentException("Message cannot be null or empty.", "message");
      }
      IDictionary<string, string> json;
      try {
        json = JsonConvert.DeserializeObject<IDictionary<string, string>>(message);
      } catch {
        throw new ArgumentException("Unable to parse json.");
      }
      if (json.ContainsKey("title")) {
        Title = json["title"];
      }
      if (json.ContainsKey("content")) {
        Content = json["content"];
      }
      if (json.ContainsKey("extra"))
      {
        Extras = JsonConvert.DeserializeObject<Dictionary<string, string>>(json["extra"]);
      } else {
        Extras = new Dictionary<string, string>();
      }
      CollapseKey = json.ContainsKey("collapse_key") ? json["collapse_key"] : null;
      AndroidNotificationId = json.ContainsKey("notification_id") ? Convert.ToInt32(json["notification_id"]) : 0; 
    }
```

I know that adding the an additional package dependency may not be the best solution for you due to potential conflicts with other clients, but this is a very good C# json parsing package.

**Comment by kolibrigames-ajimine-vahid** [comment]
Also, here is a link to the Canvas that sends the push notification:

https://dashboard-01.braze.eu/engagement/canvas/5f732975d57d0524dd65a448/5e679de743c19d0548ae31dd?isEditing=false

**Comment by radixdev** [comment]
Which platform (android/iOS) are you seeing this on?
Do you have an example campaign to reproduce this on?
What device did you see this on?
What unity SDK version are you seeing this on?

**Comment by kolibrigames-ajimine-vahid** [comment]
This is only for Android thus far.

Just create a canvas, and send a test push notification to a player, but you can test this with the sample string with the `new PushNotification(someJson)`

I see this on the current 2.5.0 version

**Comment by kolibrigames-ajimine-vahid** [comment]
This happened on a One Plus 5 A5000

**Comment by kolibrigames-ajimine-vahid** [comment]
But I was able to unit test this issue directly with c# with the `PushNotification` class with the aforementioned json string

**Comment by radixdev** [comment]
Should be resolved in 2.6.0. Please let me know if this fits your usecase.
